controller/CosmeticAPIController.py

```diff
@@ -0,0 +1,254 @@
+
+
+@app.route('/searchproduct', methods=['GET'])
+@jwt_required
+def search_product():
+    try:
+        products = ProductRepo.get_all_products(filters=request.args)
+        if len(products) > 0:
+            return jsonify(products=Product.serialize_list(products))
+        else:
+            return make_response(jsonify(response='No Products available for the given search criteria'), 200)
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
+
+
+"""
+Rest API : for add product
+Input: Allows to pass the new product details as
+                    request args or request body
+"""
+
+
+@app.route('/addproduct', methods=['POST'])
+@jwt_required
+def add_product():
+    try:
+        if len(request.args) == 0:
+            product = request.json
+        else:
+            product = json.loads(json.dumps(request.args))
+        if product is not None:
+            product_id = ProductRepo.create_product(product)
+        else:
+            return make_response(jsonify(response=f'Request data is invalid.'), 400)
+        return make_response(jsonify(response=f'Product added successfully and the product_id is {product_id} '), 201)
+    except sqlalchemy.exc.InvalidRequestError as reqerr:
+        return make_response(jsonify(error_message=str(reqerr)), 400)
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
+
+
+"""
+Rest API : for update product
+Input: product_id - Id of the product which is to be updated
+       PATCH - Updates specific fields
+       PUT   - Updates all the product data given in the request body
+               Uses default values for empty fields
+"""
+
+
+@app.route('/updateproduct/<int:product_id>', methods=['PUT', 'PATCH'])
+@jwt_required
+def update_product(product_id):
+    try:
+        logging.info(f'updateProduct: User has given input product_id as: {product_id}')
+        product = ProductRepo.getProduct(product_id)
+        if product is not None:
+            if request.method == 'PUT':
+                ProductRepo.updateallfields(product_id, request.json)
+            else:
+                fields = request.json
+                ProductRepo.update(product_id, fields)
+            return jsonify(response=f'Updated product with id {product_id} successfully')
+        else:
+            return make_response(jsonify(response=f'Request data is invalid.'), 400)
+    except sqlalchemy.exc.InvalidRequestError as error:
+        return make_response(jsonify(error_message=str(error)), 400)
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
```

Comment:
Based on the previous suggestion:
```suggestion
    logging.info(f'updateProduct: User has given input product_id as: {product_id}')
    try:
        product = ProductRepo.getProduct(product_id)
        if product is None:
            return make_response(jsonify(response=f'Request data is invalid.'), 400)
        if request.method == 'PUT':
            ProductRepo.updateallfields(product_id, request.json)
        else:
            fields = request.json
            ProductRepo.update(product_id, fields)
        return jsonify(response=f'Updated product with id {product_id} successfully')
 
    except (sqlalchemy.exc.InvalidRequestError,  exc.SQLAlchemyError) as error:
        return make_response(jsonify(error_message=str(error)), 400)
```